api/routes/product.js

```diff
@@ -1,12 +1,13 @@
 const router = require('express').Router();
 const product = require('../controllers/product');
 const validator = require('../utils/validations');
-const { check } = require('express-validator');
 
 router.get('/productsFilters', product.filterOptions);
 
 router.get('/products', product.paginatedResults);
 
+router.get('/product', product.singleProduct);
```

Comment:
Use url params for getting single item, instead of query params
```suggestion
router.get('/product/:id', product.singleProduct);
```